homeassistant/components/konnected/sensor.py

```diff
@@ -0,0 +1,124 @@
+"""
+Support for DHT and DS18B20 sensors attached to a Konnected device.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.konnected/
+"""
+import logging
+
+from homeassistant.components.konnected import (
+    DOMAIN as KONNECTED_DOMAIN, SIGNAL_DS18B20_NEW, SIGNAL_SENSOR_UPDATE)
+from homeassistant.const import (
+    CONF_DEVICES, CONF_PIN, CONF_TYPE, CONF_NAME, CONF_SENSORS,
+    DEVICE_CLASS_HUMIDITY, DEVICE_CLASS_TEMPERATURE, TEMP_FAHRENHEIT)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.util.temperature import celsius_to_fahrenheit
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['konnected']
+
+SENSOR_TYPES = {
+    DEVICE_CLASS_TEMPERATURE: ['Temperature', None],
+    DEVICE_CLASS_HUMIDITY: ['Humidity', '%']
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up sensors attached to a Konnected device."""
+    if discovery_info is None:
+        return
+
+    SENSOR_TYPES[DEVICE_CLASS_TEMPERATURE][1] = \
+        hass.config.units.temperature_unit
+
+    data = hass.data[KONNECTED_DOMAIN]
+    device_id = discovery_info['device_id']
+    sensors = []
+
+    # Initialize all DHT sensors.
+    for sensor in filter(lambda s: s[CONF_TYPE] == 'dht',
+                         data[CONF_DEVICES][device_id][CONF_SENSORS]):
+        sensors.append(
+            KonnectedSensor(device_id, sensor, DEVICE_CLASS_TEMPERATURE))
+        sensors.append(
+            KonnectedSensor(device_id, sensor, DEVICE_CLASS_HUMIDITY))
+
+    async_add_entities(sensors)
+
+    @callback
+    def async_add_ds18b20(data):
+        """Add new KonnectedSensor representing a ds18b20 sensor."""
+        async_add_entities([
+            KonnectedSensor(data.get('device_id'), data,
+                            DEVICE_CLASS_TEMPERATURE)
+        ], True)
+
+    # DS18B20 sensors entities are initialized when they report for the first
+    # time. Set up a listener for that signal from the Konnected component.
+    async_dispatcher_connect(hass, SIGNAL_DS18B20_NEW, async_add_ds18b20)
+
+
+class KonnectedSensor(Entity):
+    """Represents a Konnected DHT Sensor."""
+
+    def __init__(self, device_id, data, sensor_type):
+        """Initialize the entity for a single sensor_type."""
+        self._data = data
+        self._device_id = device_id
```

Comment:
Is that a unique ID? So you could use that for unique_id property and allow to set name over the UI.

Reply:
It's the unique id of the Konnected device, but not unique to the sensor entity. Each device may have multiple sensor entities.

Reply:
Sure. Theoretical we support the device_entry and allow to constrain like what you say. I.e. deconz.

Anyway, make it unique with `"{}-{}".format(device_id, sensor_type)` and set the unique_id property on entity.

Reply:
After that you can remove the naming property and user can set it over the UI.